tests/test/scheduler/test_mpi_world.cpp

```diff
@@ -301,6 +301,7 @@ TEST_CASE("Test send and recv on same host", "[mpi]")
         REQUIRE(status.bytesSize == messageData.size() * sizeof(int));
     }
 
+    /*
```

Comment:
By switching to `assert`-based errors, some tests that rely on `REQUIRE_THROWS` are not passing. I leave them commented here to remember to discuss about this. 

I've done some reading and it seems as though we could do [some dirty hacks](https://stackoverflow.com/a/38551114/10662021) to have assertions fails throw exceptions during testing, but not convinced.